### ISSUE TITLE
eval and exec are not yet implemented

### ISSUE THREAD

**Issue by acbart** [skulpt/skulpt#787]
It seems like eval and exec are not yet implemented. Any plans for them to be set up? Naively, given the existence of misceval, it seems like they wouldn't be too difficult. Are there roadblocks I'm missing?

**Comment by acbart** [comment]
Think it'd be more than just wrapping `Sk.importModuleInternal_ ` ?

**Comment by Dustyposa** [comment]
@acbart 
Have any idea? `Sk.importModuleInternal_` can't do this. Right?
Need locals var, global var, and a way to compile code when the script is running.
First, have any function to run a code segment?

**Comment by acbart** [comment]
Yeah, we've actually successfully gotten `exec` working in our fork. It was not easy, and has required a whole mess of changes in many places. The actual function can be seen here: https://github.com/blockpy-edu/skulpt/blob/master/src/builtin.js#L1424

But if you want to get my changes, you should probably diff against our whole skulpt directory to find out what else it depended on. It's relatively up-to-date to the master. I need to extract out a half-dozen features and bug fixes to push upstream, but I've already been moving at a break-neck speed this summer.

**Comment by acbart** [comment]
And I don't claim that this is thoroughly unit tested. I'm pretty sure there's a lot of little aspects that are not working - which is particularly troublesome since the whole point of making `exec` work was to get [Pedal](https://pedal-edu.github.io/pedal/html/index.html)'s [sandbox module](https://github.com/pedal-edu/pedal/blob/master/pedal/sandbox/sandbox.py) working, and it does some *crazy* stuff.

**Comment by Dustyposa** [comment]
Oh, thanks. Has a bit of message for me. That's look not easy as my thought. :(
I will look over your master branch.

Some students use the `eval` function, but not support in my master branch....
That's a sad news. If complete this function too hard for me, I will make a online interpreter which use python interpreter interpret the code for the present.
